runtime/permissions.rs

```diff
@@ -81,9 +116,15 @@ pub struct UnaryPermission<T: Eq + Hash> {
 }
 
 #[derive(Clone, Eq, PartialEq, Hash, Debug, Default, Deserialize)]
-pub struct ReadPermission(pub PathBuf);
+pub struct ReadDescriptor(pub PathBuf);
+
+#[derive(Clone, Eq, PartialEq, Hash, Debug, Default, Deserialize)]
+pub struct WriteDescriptor(pub PathBuf);
+
+#[derive(Clone, Eq, PartialEq, Hash, Debug, Default, Deserialize)]
+pub struct NetDescriptor(pub String, pub Option<u16>);
```

Comment:
why did you move the descriptor definitions all together? having them to their corresponding impl block seems better to me

Reply:
I had trouble finding them initially, and this was where I looked for them to be so it felt right. Probably because the impl blocks are for `UnaryPermission<_>`.

Reply:
hm, alright. there is also some impls for the NetDescriptor directly, so maybe you wanna move those up as well

Reply:
(you forgot the display impl for net descriptor)